### ISSUE TITLE
Returning None from a function with Option<i16> return type doesn't result in "undefined"

### ISSUE THREAD

**Issue by jamornsriwasansak** [rustwasm/wasm-bindgen#3133]
### Describe the Bug

Returning None from a function with Option<i16> return type doesn't result in "undefined"

### Steps to Reproduce
Build a wasm with a following function and call this function from javascript.
``` Rust
#[wasm_bindgen]
pub fn test() -> Option<i16> {
    None
}
```

If applicable, add a link to a test case (as a zip file or link to a repository we can clone).

### Expected Behavior

The function should return `undefined` when being called by javascript.

### Actual Behavior

The function returns `16777215` when being called by javascript.

**Comment by RunDevelopment** [comment]
I cannot reproduce this issue. Either this was never bug or has been fixed.

Here's what I did: I compiled the code snippet from the issue. The relevant part of the JS glue code looks like this:

```js
/**
 * @returns {number | undefined}
 */
export function test() {
    const ret = wasm.test();
    return ret === 0xFFFFFF ? undefined : ret;
}
```

`0xFFFFFF == 16777215`, so my guess is that they used the exported wasm function directly without the JS glue code.

To verify this, I loaded the wasm module with deno:

```js
> import * as m from "./test.js"
undefined
> var wasm = await WebAssembly.instantiate(Deno.readFileSync("./test.wasm"))
undefined
> m.__wbg_set_wasm(wasm.instance.exports)
undefined
> m.test()
undefined
> wasm.instance.exports.test()
16777215
```

Works as expected.

For more context, `0xFFFFFF` is the sentinile value wasm bindgen uses for the options of all primitive values <32 bits, so `bool`, `u8`, `i8`, `u16`, `i16`, and `char`. Given that there are no issues for those types, I suspect that this issue was caused by user error.

@daxpedda I think this issue can be closed.
